frontend/src/lib/components/PayGateMini/payGateMiniLogic.tsx

```diff
@@ -86,10 +86,13 @@ export const payGateMiniLogic = kea<payGateMiniLogicType>([
         ],
         nextPlanWithFeature: [
             (s) => [s.productWithFeature],
-            (productWithFeature) =>
-                productWithFeature?.plans.find(
-                    (plan) => plan.features?.some((f) => f.key === props.featureKey) && !plan.current_plan
-                ),
+            (productWithFeature) => {
+                const currentPlanIndex = productWithFeature?.plans.findIndex((plan) => plan.current_plan)
+                if (!currentPlanIndex) {
+                    return null
+                }
+                return productWithFeature?.plans[currentPlanIndex + 1]
```

Comment:
What specifically? I check for `currentPlanIndex,` and if plans don't have anything at that index, it will return undefined.

Example
```
> arr = ['0', '1', '2', '3']
[ '0', '1', '2', '3' ]
> arr[2]
'2'
> arr[5]
undefined
```